src/toil/common.py

```diff
@@ -904,3 +908,50 @@ def cacheDirName(workflowID):
     :return: Name of the cache directory.
     """
     return 'cache-' + workflowID
+
+
+def getSizeOfDirectory(dirToSizeUp):
+    '''
+    This method will walk through a directory and return the cumulative filesize in bytes of all
+    the files in the directory and its subdirectories.
+
+    :param dirToSizeUp: path to the workflow working directory
+    :return: cumulative size in bytes of all files in the directory
+    :rtype: float
+    '''
+    totalSize = 0
+    # The value from running stat on each linked file is equal. To prevent the same file
+    # from being counted multiple times, we save the inodes of files that have more than one
+    # nlink associated with them.
+    seenInodes = set()
+    for dirPath, dirNames, fileNames in os.walk(dirToSizeUp):
+        folderSize = _getDirSize(dirPath, fileNames, seenInodes)
+        totalSize += folderSize
+    return totalSize
+
+
+def _getDirSize(dirPath, fileNames, seenInodes):
+    '''
+    Returns the size of the directory including all linked files provided their inode was not in
+    linkedFileInodes.
+
+    :param str dirPath: Dir path
+    :param list fileNames: Files in the directory
+    :param set seenInodes: set of inode numbers of files that have multiple links, and have already
+           been acccounted for.
+    :return: Size of the directory in bytes
+    :rtype: float
```

Comment:
Why float? You can't have half a byte. And folderSize is initialized to an int and only ints are added so I don't see where that value gets converted to a float.